### PR TITLE
Notes for 19602

### PR DESCRIPTION
**Pull request by achow101**

### CODE REVIEW

File: _posts/2022-03-09-#19602.md [hunk 1]
```diff
@@ -10,15 +10,60 @@ status: upcoming
 commit:
```
Comment (glozow) [inline _posts/2022-03-09-#19602.md:10]: Pushed a tag for bfd38435ad, feel free to paste here

File: _posts/2022-03-09-#19602.md [hunk 2]
```diff
@@ -10,15 +10,60 @@ status: upcoming
...
+   (Hint: Look at [`LegacyScriptPubKeyMan::IsMine`](https://github.com/bitcoin/bitcoin/blob/267917f5632a99bb51fc3fe516d8308e79d31ed1/src/wallet/scriptpubkeyman.cpp#L215))?
+
+5. Under what circumstances will a Legacy wallet watch for a multisig?
+
+6. Home come `LegacyScriptPubKeyMan::GetAllScriptPubKeys()` is needed
+   (Hint: Consider how [`DescriptorScriptPubKeyMan::IsMine`](https://github.com/bitcoin/bitcoin/blob/267917f5632a99bb51fc3fe516d8308e79d31ed1/src/wallet/scriptpubkeyman.cpp#L1705) works))?
+
+7. How come the HD seed is not ignored when looking at all of the keys in a Legacy wallet?
+
+8. What happens to watch-only things that were in the Legacy wallet?
```
Comment (glozow) [inline _posts/2022-03-09-#19602.md:64]: Instead of "things," maybe specify the objects?

**Review by glozow — COMMENTED** [review]
Thanks @achow101! Just a few suggestions.

File: _posts/2022-03-09-#19602.md (hunk 1 above)
Comment (achow101) [reply _posts/2022-03-09-#19602.md:10]: The current latest is df1dca43c277704d3959253037cba651ba2c0b46, should I still use the one that is tagged?

File: _posts/2022-03-09-#19602.md [hunk 3]
```diff
@@ -10,15 +10,60 @@ status: upcoming
...
+
+* The old type of wallets are referred to as Legacy wallets and they are slowly being deprecated
+  and removed. However there needs to be a way to turn a Legacy wallet into a Descriptor wallet
+  so that users do not find themselves unable to use their wallet.
+
+* In addition to changing how keys and scripts are tracked, Descriptor wallets also redefine the
+  concept (and implementation) of `IsMine`. Instead of allowing both private keys and watch-only addresses in a single
+  wallet, Descriptor wallets can either always have private keys, or never have private keys.
+
+* This PR implements an RPC which migrates a Legacy wallet into a descriptor wallet by computing
```
[the remark this answers is not in the document]
Comment (achow101) [reply _posts/2022-03-09-#19602.md]: Done

File: _posts/2022-03-09-#19602.md [hunk 4]
```diff
@@ -10,15 +10,60 @@ status: upcoming
...
+   * A Legacy wallet can be created with `bitcoin-cli -named createwallet wallet_name=<name> descriptors=false`
+
+3. In your own words, briefly describe the migration process.
+
+4. What kind of scripts will a Legacy wallet consider to belong to it for a single key
+   (Hint: Look at [`LegacyScriptPubKeyMan::IsMine`](https://github.com/bitcoin/bitcoin/blob/267917f5632a99bb51fc3fe516d8308e79d31ed1/src/wallet/scriptpubkeyman.cpp#L215))?
+
+5. Under what circumstances will a Legacy wallet watch for a multisig?
+
+6. Home come `LegacyScriptPubKeyMan::GetAllScriptPubKeys()` is needed
```
[the remark this answers is not in the document]
Comment (achow101) [reply _posts/2022-03-09-#19602.md]: Done

File: _posts/2022-03-09-#19602.md [hunk 5]
```diff
@@ -10,15 +10,60 @@ status: upcoming
...
+   (Hint: Consider how [`DescriptorScriptPubKeyMan::IsMine`](https://github.com/bitcoin/bitcoin/blob/267917f5632a99bb51fc3fe516d8308e79d31ed1/src/wallet/scriptpubkeyman.cpp#L1705) works))?
+
+7. How come the HD seed is not ignored when looking at all of the keys in a Legacy wallet?
```
[the remark this answers is not in the document]
Comment (achow101) [reply _posts/2022-03-09-#19602.md]: Done

File: _posts/2022-03-09-#19602.md (hunk 2 above)
Comment (achow101) [reply _posts/2022-03-09-#19602.md]: Done

**Comment by glozow** [comment]
Beautiful, thank you @achow101! Looking forward to it!

**Pull request merged**